packages/forklift-console-plugin/src/modules/Plans/views/details/tabs/Mappings/PlanMappings.tsx

```diff
@@ -105,9 +183,25 @@ const PlanMappingsInitSection: React.FC<PlanMappingsInitSectionProps> = (props)
   const [targetStorages, targetStoragesLoading, targetStoragesError] =
     useOpenShiftStorages(targetProvider);
 
+  const [vmData] = useInventoryVms(
+    { provider: sourceProvider },
+    providersLoaded,
+    providersLoadError,
+  );
+  const migratedVmIds = plan?.status?.migration?.vms?.reduce((migrated, vm) => {
+    if (getVMMigrationStatus(vm) === 'Succeeded') {
+      migrated.push(vm.id);
+    }
+    return migrated;
+  }, []);
+  const migratedVms = vmData.filter((vm) => migratedVmIds?.includes(vm.vm.id));
+  const networkIdsUsedByMigratedVms =
+    sourceProvider?.spec?.type !== 'ovirt' ? getNetworksUsedBySelectedVms(migratedVms, []) : [];
+  const storageIdsUsedByMigratedVms = ['ovirt', 'openstack'].includes(sourceProvider?.spec?.type)
+    ? []
+    : getStoragesUsedBySelectedVms({}, migratedVms, []);
```

Comment:
useMemo to prevent rendering and exporting to a util file

Reply:
I applied useMemo